src/BookShelf1L.java

```diff
@@ -0,0 +1,271 @@
+
+    /**
+     * Default size of hash table.
+     */
+    private static final int DEFAULT_COLUMN_SIZE = 100;
+    /**
+     * Users bookShelf.
+     */
+    private Map<String, Map<String, String>>[][] bookShelf;
+    /*
+     * List of Books Currently In Progress
+     */
+    /*
+     * Keeps track of the number of books in the shelf
+     */
+    private int numberOfBooksInShelf = 0;
+    /*
+     * Stores the books that the user is currently reading
+     */
+    private Stack<Map<String, Map<String, String>>> listOfBooksInProgress;
+    /*
+     * Number of Books the User Wants to Read
+     */
+    private int numberOfBooksToRead;
+    /*
+     * Number of Books the user has read
+     */
+    private in numberOfBooksRead;
+    /*
+     * Number of books the user is currently reading
+     */
+    private int numberOfBooksInProgress;
+    /*
+     * Whether the user has reached their goal
+     */
+    private boolean goalReached;
+
+    /**
+     * Creator of initial representation.
+     */
+    @SuppressWarnings("unchecked")
+    private void createNewRep(int rowSize, int colSize, int numBooksToRead) {
+
+        this.bookShelf = new Map<String, Map<String, String>>[rowSize][colSize];
+        for (int i = 0; i < this.bookShelf.length; i++) {
+            for (int j = 0; j < this.bookShelf[0].length; j++) {
+                this.bookShelf[i][j] = new Map1L<String, Map<String, String>>();
+            }
+        }
+        this.numberOfBooksRead = numBooksToRead;
+
+    }
+    /*
+     * Constructors -----------------------------------------------------------
+     */
+
+    /**
+     * No-argument constructor.
+     */
+    public BookShelf1L() {
+
+        this.createNewRep(DEFAULT_ROW_SIZE, DEFAULT_COLUMN_SIZE, 0);
+
+    }
+
+    /**
+     * Constructor creates a array based on specified rows and columns and
+     * initializes the number of books the user wants to read
+     *
+     * @param rowSize
+     *            length of the rows of bookShelf
+     * @param colSize
+     *            length of columns of bookShelf
+     * @ensures this = {}
+     */
+    public BookShelf1L(int rowSize, int colSize, int numBooksToRead) {
+
+        this.createNewRep(rowSize, colSize, numBooksToRead);
+
+    }
+
+    /*
+     * TransferFrom
+     */
+    @Override
+    public final void transferFrom(BookShelf source) {
+        assert source != null : "Violation of: source is not null";
+        assert source != this : "Violation of: source is not this";
+        assert source instanceof BookShelf1L<?, ?> : ""
+                + "Violation of: source is of dynamic type BookShelf1L<?,?>";
+
+        BookShelf1L localSource = (BookShelf1L) source;
+        this.bookShelf = localSource.bookShelf;
+        this.numberOfBooksToRead = localSource.numberOfBooksToRead;
+        localSource.createNewRep(DEFAULT_ROW_SIZE, DEFAULT_COLUMN_SIZE, 0);
+    }
+
+    public Map<String, Map<String, String>>[][] displayShelf() {
```

Comment:
I haven't looked recently, but a lot of these methods seem like they should be secondary methods (i.e., implemented in the abstract class).